scipy/conftest.py

```diff
@@ -200,7 +200,11 @@ def skip_if_array_api(xp, request):
             elif xp.__name__ == 'torch':
                 if 'cpu' not in torch.empty(0).device.type:
                     pytest.skip(reason=reason)
-    if backends is not None:
+    if "all" in backends:
+        reason = "do not run at all with SCIPY_ARRAY_API=1"
+        reason = kwargs.get("reasons", reason)
+        pytest.skip(reason=reason)
+    elif backends is not None:
```

Comment:
up until now I have opted for simply `@pytest.mark.skipif(SCIPY_ARRAY_API)` - do you think it is better to include skipping 'all' in this decorator? I would say not, since this type of skip should only be used for things which we are planning to get rid of by the time this stops being experimental, like `numpy.matrix`.

Reply:
You're probably right. I misunderstood the purpose of the decorator initially; for instance, I expected `pytest.mark.skip_if_array_api()` to skip all, not skip none. Anyway, now that I understand that the purpose is to skip specific backends, I can see that `"all"` is not needed.

Perhaps if this were named `@pytest.mark.skip_backends`, I would have understood the intent better?

Reply:
> @pytest.mark.skip_backends

I also prefer this name. However, I think I am biased coming with the knowledge of the array API context. In code where there are other 'backends' (uarray, compiled code etc.), I think that could cause more confusion than it helps avoid.

The important thing is to make sure that http://scipy.github.io/devdocs/dev/api-dev/array_api.html#adding-tests is clear - let me know if there are any improvements I could make there.